File: Lista_1/Python/1.py
vetor = [0,1,2,3,11,5,6]

def buscaRecursiva(vetor, n, indice = 0):
    if(indice >= len(vetor)): #caso base / restrição
        return False
    elif(n == vetor[indice]): # achou o elemento
        return indice
    else:
        return buscaRecursiva(vetor, n, indice + 1) # chama recursivamente incrementando o índice

# exemplo de uso
resultado = buscaRecursiva(vetor, 11)
print(resultado)

# A complexidade de tempo desta função é O(n), ou seja, proporcional ao tamanho do vetor (pior caso).
# A complexidade de memória é O(1), visto que independe da entrada, somente de variáveis locais.

File: Lista_1/C/1.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    
    int procuraElemento(int array[], int n, int indice, int tamanho){
        if(indice == tamanho){ // caso base
            return -1;
        }
        else if(array[indice] == n){
            return indice;
        }
        return procuraElemento(array, n, indice + 1, tamanho);
    }
    
    
    int array[] = {0,1,2,3,4,5,6};
    int n = 4;
    int tamanho = sizeof(array) / sizeof(array[0]); // calcula o tamanho do vetor
    printf("%d", procuraElemento(array, n, 0, tamanho));
    return 0;
}

